client-spec.md

```diff
@@ -0,0 +1,275 @@
+
+### 4.1. Inputs
+
+The Verifier performs verification according to its policy based on the following inputs:
+
+* The artifact.  
+* Verification materials (possibly in the the `Bundle` format ([definition](https://github.com/sigstore/protobuf-specs/blob/88c45b0ab8c3781a118be6339f443d8c277c0126/protos/sigstore\_bundle.proto\#L61-L77))):  
+  * Leaf certificate  
+    * When used with the Public Good Instance, only the leaf is necessary. Other Sigstore instances (such as private instances) may require one or more intermediates as well, if those intermediates are not listed in the independent root of trust.  
+  * Signature.  
+  * Additional payload metadata.  
+  * Timestamping response.  
+  * Transparency Service `LogEntry` ([definition](https://github.com/sigstore/rekor/blob/0a3f871c077eb708f2ffcc382d0a2104b887f5e1/openapi.yaml\#L423-L464)).  
+* Root key material for Sigstore infrastructure (from the policy).
+
+The distribution of these inputs is out-of-scope for this document.
+
+#### 4.1.1. Recommended discovery order for verification materials
+
+For Sigstore clients that expose a command-line interface, the following discovery order is RECOMMENDED:
+
+1. Use whatever verification materials are supplied explicitly by the user. For example, if the client has flags and/or environment variables for configuring bundles and/or detached verification materials, these should take precedence over any implicitly discovered materials.  
+2. If no explicit inputs are given: for a given file `input`, attempt to discover `{input}.sigstore.json`. If `{input}.sigstore.json` is present, attempt to use it for verification.  
+3. If `{input}.sigstore.json` is not present, attempt to discover `{input}.sigstore` and use it for verification.
+
+### 4.2. Establishing a Time for the Signature
+
+First, establish a time for the signature. This timestamp is required to validate the certificate chain, so this step comes first.
+
+#### 4.2.1. Timestamping Service
+
+If the verification policy uses the Timestamping Service, the Verifier MUST verify the timestamping response using the Timestamping Service root key material, as described in [Spec: Timestamping Service](https://docs.google.com/document/u/0/d/1FoRHXejIhXwEai0RS3iRsN1HfCV16fJOp582Vl8KA7A/edit), with the raw bytes of the signature as the timestamped data. The Verifier MUST then extract a timestamp from the timestamping response. If verification or timestamp parsing fails, the Verifier MUST abort.
```

Comment:
Left a comment about this in the doc, but Verifiers shouldn't abort for this or the next step imo. Verifiers should gather as many timestamps as they can successfully validate, and use each when verifying the certificate.

Reply:
Tracked as #18 (and also linked in top-level issue #12 )